### PR TITLE
Sentence transformers in the Hub

### PR DESCRIPTION
**Pull request by osanseviero**
There are still a couple of TODOs and some hacks. Here are some screenshots

Screen Shot 2021-06-15 at 4 08 13 PM
Screen Shot 2021-06-15 at 4 08 20 PM
Screen Shot 2021-06-15 at 4 08 27 PM
Screen Shot 2021-06-15 at 4 08 31 PM
Screen Shot 2021-06-15 at 4 08 37 PM
Screen Shot 2021-06-15 at 4 08 42 PM
Screen Shot 2021-06-15 at 4 08 45 PM

### CODE REVIEW

**Comment by osanseviero** [comment]
@patrickvonplaten in case you have a free cycle to take a look :)

File: sentence-transformers-in-the-hub.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    <a href="/nreimers">
+        <img class="avatar avatar-user" src="https://aeiljuispo.cloudimg.io/v7/https://aeiljuispo.cloudimg.io/v7/https://aeiljuispo.cloudimg.io/v7/https://s3.amazonaws.com/moonup/production/uploads/1596792577829-5eff4688ff69163f6f59e66c.jpeg?w=200&h=200&f=face" title="Gravatar">
+        <div class="bfc">
+            <code>nreimers</code>
+            <span class="fullname">Nils Reimers</span>
+        </div>
+    </a>
+</div>
+
+Over the past few weeks, we've built collaborations with many Open Source frameworks in the machine mearning ecosystem. One that gets us particularly excited is Sentence Transformers.
```
Comment (patrickvonplaten) [inline sentence-transformers-in-the-hub.md:34]:
```suggestion
Over the past few weeks, we've built collaborations with many Open Source frameworks in the machine learning ecosystem. One that gets us particularly excited is Sentence Transformers.
```

File: sentence-transformers-in-the-hub.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+[Sentence Transformers](https://github.com/UKPLab/sentence-transformers) is a framework for sentence, paragraph and image embeddings. This allows to derive semantically meaningful embeddings (1) which is useful for very interesting applications, such as semantic search or mutli-lingual zero shot classification. As part of Sentence Transformers [v2 release](TODO add link), there are a lot of cool new features:
```
Comment (patrickvonplaten) [inline sentence-transformers-in-the-hub.md:36]: I'd remove the "very interesting" part.

File: sentence-transformers-in-the-hub.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+- Sharing your models in the 🤗 Hub easily.
+- Widgets and Inference API for sentence embeddings and sentence similarity.
+- Better sentence-embeddings models available ([benchmark](https://www.sbert.net/docs/pretrained_models.html#sentence-embedding-models) and [models](https://huggingface.co/sentence-transformers) in the 🤗 Hub).
+- Learning to rank (LTR) functions for cross-encoder training.
```
Comment (patrickvonplaten) [inline sentence-transformers-in-the-hub.md:41]: Maybe a link here that explains what LTR is?

Comment (nreimers) [reply sentence-transformers-in-the-hub.md:41]: I decided to move the LTR functions to version 2.1

I want to make a big special on LTR with educational material, which LTR methods exists, how they can be used and which LTR method yields the best result. But this requires still some time & experiments, hence, I moved it to version 2.1 of sentence-transformers.

File: sentence-transformers-in-the-hub.md (hunk 2 above)
Comment (osanseviero) [reply sentence-transformers-in-the-hub.md]: Done :)

File: sentence-transformers-in-the-hub.md (hunk 3 above)
Comment (osanseviero) [reply sentence-transformers-in-the-hub.md]: Thanks, I just removed that point

**Review by patrickvonplaten — APPROVED** [review]

**Comment by osanseviero** [comment]
@gary149 do you think you could help us with a thumbnail for this and other sentence transformers announcements? It could be something like Sentence Transformers + Hugging Face like we've done for other collaborations.

CC @julien-c in case you want to check out the blog post. Sorry for the ugly hacks for the widget, it's some technical debt that I think it's ok until we improve a bit the widgets situation. 😅 

@nreimers I updated the sentence similarity widget to use a sentence similarity widget. I also changed to the new `save_to_hub` method. I think everything is pretty much ready, we're just missing the link to the actual release 😄

**Comment by nreimers** [comment]
@osanseviero 
Here the link to the release:
https://github.com/UKPLab/sentence-transformers/releases/tag/v2.0.0

Release notes will be added tomorrow.

File: sentence-transformers-in-the-hub.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+import json
+import requests
+
+API_URL = "https://api-inference.huggingface.co/models/sentence-transformers/paraphrase-MiniLM-L6-v2"
+headers = {"Authorization": "Bearer YOUR_TOKEN"}
+
+def query(payload):
+	data = json.dumps(payload)
+	response = requests.request("POST", API_URL, headers=headers, data=data)
+	return json.loads(response.content.decode("utf-8"))
```
Comment (julien-c) [inline sentence-transformers-in-the-hub.md:87]:
```suggestion
	response = requests.post(API_URL, headers=headers, json=payload)
	return response.json()
```

Comment (julien-c) [reply sentence-transformers-in-the-hub.md:87]: (yeah i know we need to update the inference API modal, not sure why it was written like this for the first place)

File: sentence-transformers-in-the-hub.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+model.save_to_hub("my_new_model")
+```
+
+Now you will have a [repository](https://huggingface.co/osanseviero/my_new_model) in the Hub which hosts your model. A model card was automatically created. It describes the architecture by listing the layers and shows how to use the model with both `Sentence Transformers` and `🤗 Transformers`. You can also try out the widget and use the Inference API straight away!
+
+If this was not exciting enough, your models will also be easily discoverable by [filtering for all](https://huggingface.co/models?filter=sentence-transformers) `Sentence Transformers` models.
+
+## What's next?
+
+Moving forward, we want to make this integration even more useful. In our roadmap, we expect training and evaluation data to be included in the automatically created model card.
```
Comment (julien-c) [inline sentence-transformers-in-the-hub.md:121]:
```suggestion
Moving forward, we want to make this integration even more useful. In our roadmap, we expect training and evaluation data to be included in the automatically created model card, like is the case in `transformers` from version `v4.8`.
```

**Comment by julien-c** [comment]
looks good!

> Sorry for the ugly hacks for the widget, it's some technical debt that I think it's ok until we improve a bit the widgets situation. 😅

I think that's totally fine. We have a way to render (just) the widget server side so we could have hacked together something outside of the .md, but I think this works

**Comment by LysandreJik** [comment]
:tada:

**Pull request merged**